Objects/rangeobject.c

```diff
@@ -409,14 +409,17 @@ compute_slice(rangeobject *r, PyObject *_slice)
     PyObject *substart = NULL, *substop = NULL, *substep = NULL;
     int error;
 
-    if (slice->start == Py_None && slice->stop == Py_None && slice->step == Py_None) {
-        return Py_NewRef(r);
-    }
-
     error = _PySlice_GetLongIndices(slice, r->length, &start, &stop, &step);
     if (error == -1)
         return NULL;
 
+    if (start == _PyLong_GetZero()
+        && step == _PyLong_GetOne()
+        && (slice->stop == Py_None || PyObject_RichCompareBool(stop, r->length, Py_EQ) == 1))
+    {
+        return Py_NewRef(r);
+    }
```

Comment:
`PyObject_RichCompareBool` can raise an error. Handling it would then make the code more complex. So I'm rather against that change as it's not so trivial.

Also, `start == _PyLong_GetZero()` is not correct because it's not guaranteed that 0 will always be immortal. I actually thought that we had the indices as C integers and not Python ones, but it appears that it's not the case.